apps/src/applab/commands.js

```diff
@@ -1381,7 +1384,9 @@ applabCommands.startWebRequest = function (opts) {
   apiValidateType(opts, 'startWebRequest', 'callback', opts.func, 'function');
   var req = new XMLHttpRequest();
   req.onreadystatechange = applabCommands.onHttpRequestEvent.bind(req, opts);
-  req.open('GET', opts.url, true);
+  var url = XHR_PROXY_PATH + '?u=' + encodeURIComponent(opts.url.replace(/ /g, '%20')) +
```

Comment:
Perhaps this is a heuristic to try to patch up invalid URLs?  No valid URL should contain a space. 

Most programming language libraries (as opposed to user agents) would simply reject such invalid non-URLs.  I think that's the simplest and cleanest approach.  If we do want to apply such heuristics because our developers are beginning programmers, it's probably better to do it in the calling code rather than baking it into the proxy.  (There are other reserved characters that could be misused, of course, though perhaps they aren't frequently misused enough to bother with heuristics.)